Source/DafnyCore/Verifier/ProofObligationDescription.cs

```diff
@@ -794,14 +796,27 @@ public class ForallLHSUnique : ProofObligationDescriptionWithNoExpr {
   public override string ShortDescription => "forall bound unique";
 }
 
-public class ElementInDomain : ProofObligationDescriptionWithNoExpr {
+public class ElementInDomain : ProofObligationDescription {
+  private readonly Expression sequence;
+  private readonly Expression index;
   public override string SuccessDescription =>
     "element is in domain";
 
   public override string FailureDescription =>
     "element might not be in domain";
 
   public override string ShortDescription => "element in domain";
+
+  public ElementInDomain(Expression sequence, Expression index) {
+    this.sequence = sequence;
+    this.index = index;
+  }
+  public override Expression GetAssertedExpr(DafnyOptions options) {
```

Comment:
More about the previous change, but can you add a comment to `GetAssertedExpr` in the supertype about whether the result needs to be resolved or not?